src/Microsoft.AspNetCore.SignalR/SignalRAppBuilderExtensions.cs

```diff
@@ -3,13 +3,20 @@
 
 using System;
 using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.AspNetCore.Builder
 {
     public static class SignalRAppBuilderExtensions
     {
         public static IApplicationBuilder UseSignalR(this IApplicationBuilder app, Action<HubRouteBuilder> configure)
         {
+            var marker = app.ApplicationServices.GetService(typeof(SignalRMarkerService));
+            if (marker == null)
+            {
+                throw new InvalidOperationException("The SignalR service needs to be registered");
```

Comment:
Say something about adding SignalR? Doesn’t mvc do that?

Reply:
Yeah, I'll make this more similar to what they have